File: csharp/FlatWiki/FlatWiki/FlatWiki.cs
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace FlatWiki
{


public class FlatWiki {
    private readonly string inputDirectory;
    private readonly OutputDirectory outputDirectory;

	
    public static void Main(params string[] args)  {
        new FlatWiki(args[0], new OutputDirectory(args[1])).Run();
    }

    public FlatWiki(string inputDirectory, OutputDirectory outputDirectory) {
        this.inputDirectory = inputDirectory;
        this.outputDirectory = outputDirectory;
    }

    private void Run() {
        foreach (var filename in ListInputFiles())
        {
        	var pageName = filename.Substring(0, filename.Length -".wiki".Length);
            WikiPage page = new WikiPage(pageName, ReadInputFile(filename), ModifiedTime(filename));
            outputDirectory.WriteFile(page.GetOutputFilename(), page.AsHtml());
        }
    }

    // TODO extract a delegate for inputDirectory business?
    private IEnumerable<string> ListInputFiles() {
			return Directory.EnumerateFiles(inputDirectory, "*.wiki").Select(f => new FileInfo(f).Name);
    }

    private long ModifiedTime(string filename) {
        return File.GetLastWriteTime(InputFile(filename)).Ticks;
    }

    private string ReadInputFile(string filename)  {
        return File.ReadAllText(InputFile(filename));
    }

    private string InputFile(string filename) {
        return inputDirectory + '\\' + filename;
    }
}

}

File: java/src/main/java/com/cyrusinnovation/flatwiki/FlatWiki.java
package com.cyrusinnovation.flatwiki;

import org.apache.commons.io.FileUtils;
import org.apache.commons.io.filefilter.SuffixFileFilter;

import java.io.*;

public class FlatWiki {
    private final File inputDirectory;
    private final OutputDirectory outputDirectory;

    public static void main(String... args) throws IOException {
        File inputDirectory = new File(args[0]);
        OutputDirectory outputDirectory = new OutputDirectory(args[1]);
        new FlatWiki(inputDirectory, outputDirectory).run();
    }

    public FlatWiki(File inputDirectory, OutputDirectory outputDirectory) {
        this.inputDirectory = inputDirectory;
        this.outputDirectory = outputDirectory;
    }

    private void run() throws IOException {
        for (String filename : listInputFiles()) {
            String pageName = filename.replaceFirst("\\.wiki$", "");
            WikiPage page = new WikiPage(pageName, readInputFile(filename), modifiedTime(filename));
            outputDirectory.writeFile(page.getOutputFilename(), page.asHtml());
        }
    }

    // TODO extract a delegate for inputDirectory business?
    private String[] listInputFiles() {
        return inputDirectory.list(new SuffixFileFilter(".wiki"));
    }

    private long modifiedTime(String filename) {
        return inputFile(filename).lastModified();
    }

    private String readInputFile(String filename) throws IOException {
        return FileUtils.readFileToString(inputFile(filename));
    }

    private File inputFile(String filename) {
        return new File(inputDirectory, filename);
    }
}
